### ISSUE TITLE
PagePermissionTester can_view_revisions only controls "History" link in the More buttons.

### ISSUE THREAD

**Issue by cnk** [wagtail/wagtail#10892]
![Screenshot 2023-09-13 at 5 59 30 PM](https://github.com/wagtail/wagtail/assets/5952/f11b2b29-b6a5-437d-9bc8-6e019d25768d)

### Issue Summary

I don't want a certain class of users to see the Site/Page history pages so I am monkey patching the "can_view_revisions" method of PagePermissionTester. When I set it to False, that removes the "history" link from the [page explorer's "More" actions](https://github.com/wagtail/wagtail/blob/ba9f7c898f6d9080daa6dd87100b96e8c6651355/wagtail/admin/wagtail_hooks.py#L349) - but it does not remove the "View History" link from the workflow side panel, nor from the links that open side panels. 

For the "View History" link, the template checks [if the history_url exists](https://github.com/wagtail/wagtail/blob/ba9f7c898f6d9080daa6dd87100b96e8c6651355/wagtail/admin/templates/wagtailadmin/shared/side_panels/includes/status/workflow.html#L111), so one could add to that if - or make the [get_context_data](https://github.com/wagtail/wagtail/blob/ba9f7c898f6d9080daa6dd87100b96e8c6651355/wagtail/admin/ui/side_panels.py#L255) check before adding history_url. 

I am not sure where the clock icon + link to the page history comes from in the header. That will require some investigation.

### Steps to Reproduce

In a new or existing project, monkey patch can_view_revisions to return false. For example: 

```
def patched_can_view_revisions(self):
    # BEGIN PATCH 
    return False
    # END PATCH
    return not self.page_is_root

from wagtail.models import PagePermissionTester
PagePermissionTester.can_view_revisions = patched_can_view_revisions
```

I have an entire app for patches so that's where I put it. But for reproducing the issue, it is probably sufficient to place this in a file that you then import from an `apps.py` file in one of your site's apps.  I don't see any apps.py files in Bakery demo but perhaps add one just to be sure the patch gets applied?

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

I don't have a Snippet model that uses RevisionMixin, so I am not sure if this is a consideration for Snippet history or not. 

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes / no) No - but but given the code referenced above, I am fairly sure this will be reproducible on current main branch. 

### Technical details

-   Python version: Python 3.10.11
-   Django version: 3.2.20
-   Wagtail version: 5.1.1
-   Browser version: any. I have tried Chrome and Firefox

**Comment by laymonage** [comment]
> I am not sure where the clock icon + link to the page history comes from in the header. That will require some investigation.

That comes from these templates:

https://github.com/wagtail/wagtail/blob/27a8f85e1ec798eacc4f1c00a4885c7c433f1407/wagtail/admin/templates/wagtailadmin/shared/headers/page_edit_header.html#L20-L31

https://github.com/wagtail/wagtail/blob/27a8f85e1ec798eacc4f1c00a4885c7c433f1407/wagtail/admin/templates/wagtailadmin/pages/page_listing_header.html#L19-L30

I agree that the check should utilise the existing method in the `PagePermissionTester` to be consistent, but ideally the checks should also be done on the view itself, as otherwise the view would still be accessible via the URL. That said, before we make changes that are `PagePermissionTester`-related, personally I'd like us to flesh out how it sits with our permission policies.

> I don't have a Snippet model that uses `RevisionMixin`, so I am not sure if this is a consideration for Snippet history or not.

For snippets, historically (no pun intended) the history view has existed since before `RevisionMixin` was added, and the view would just show you a list of log entries for the snippet. Given that snippet doesn't have a `PermissionTester` class, this issue doesn't really apply.

That said, I was exploring whether we should add a generic `PermissionTester` in #10578 and I wrote some thoughts on the permission policies situation in the PR description. If we proceed with the generic approach, we'd probably want to rename the method to something like `can_view_history` for the history view specifically, and either remove `can_view_revisions` or use it for e.g. previewing revisions.
